File: generate_gallery_code.js
const fs = require('fs');
const path = require('path');

const assetsDir = 'c:\\Users\\hp\\Jsroom-\\src\\Assets';
const files = fs.readdirSync(assetsDir);

// Filter for images and videos
const validExtensions = ['.jpg', '.png', '.MOV', '.svg'];
const targetFiles = files.filter(f => validExtensions.some(ext => f.endsWith(ext)));

// Generate imports
const imports = [];
const arrayItems = [];

targetFiles.forEach((f, i) => {
  const varName = `asset_${i}`;
  const sanitizedFileName = f.replace(/ /g, '_').replace(/\./g, '_');
  const importName = `asset_${sanitizedFileName}_${i}`;
  imports.push(`import ${importName} from '../Assets/${f}';`);
  
  let category = "rooms";
  if (f.toLowerCase().includes('vid') || f.endsWith('.MOV')) {
    category = "tour";
  } else if (i % 3 === 0) {
    category = "amenities";
  } else if (i % 3 === 1) {
    category = "exterior";
  }
  
  const type = f.endsWith('.MOV') ? 'video' : 'image';
  
  arrayItems.push(`  {
    id: ${i + 1},
    url: ${importName},
    category: '${category}',
    title: 'JS Room Visual ${i + 1}',
    description: 'Breathtaking view of our luxury JS Rooms ${type}.',
    featured: ${i < 8 ? "true" : "false"},
    type: '${type}'
  }`);
});

const output = `${imports.join('\n')}\n\nconst galleryImages = [\n${arrayItems.join(',\n')}\n];\n`;
fs.writeFileSync('c:\\Users\\hp\\Jsroom-\\gallery_code.txt', output);
console.log('Gallery code generated in gallery_code.txt');

File: generate_gallery_code.py
import os

assets_dir = r"c:\Users\hp\Jsroom-\src\Assets"
files = [f for f in os.listdir(assets_dir) if os.path.isfile(os.path.join(assets_dir, f))]

# Filter for images and videos
valid_extensions = ('.jpg', '.png', '.MOV', '.svg')
files = [f for f in files if f.endswith(valid_extensions)]

# Generate imports
imports = []
array_items = []

for i, f in enumerate(files):
    var_name = f"asset_{i}"
    imports.append(f"import {var_name} from '../Assets/{f}';")
    
    category = "rooms"
    if "MOV" in f:
        category = "tour"
    
    array_items.append(f"""  {{
    id: {i+1},
    url: {var_name},
    category: '{category}',
    title: '{f}',
    description: 'JS ROOMS Visual Display',
    featured: {"true" if i < 10 else "false"},
    type: '{"video" if "MOV" in f else "image"}'
  }},""")

with open(r"c:\Users\hp\Jsroom-\gallery_code.txt", "w") as out:
    out.write("\n".join(imports))
    out.write("\n\nconst galleryImages = [\n")
    out.write("\n".join(array_items))
    out.write("\n];\n")
